src/main/java/Tree/BinarySearchTree.java

```diff
@@ -63,25 +63,32 @@ public T getValue(int key) {
     protected BinaryNode<T> getSuccessor (BinaryNode<T> node) {
         assert node == null : "Input node must not be null";
 
-        node = node.getRightNode();
-        while (node.getLeftNode() != null) {
-            node = node.getLeftNode();
+        if (node.getRightNode() != null) {
```

Comment:
why is this change necessary?

Reply:
when node = null, node.getLeftNode() call produces nullpointexception